components/script/dom/gpuqueue.rs

```diff
@@ -69,4 +78,105 @@ impl GPUQueueMethods for GPUQueue {
             })
             .unwrap();
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpuqueue-writebuffer
+    #[allow(unsafe_code)]
+    fn WriteBuffer(
+        &self,
+        buffer: &GPUBuffer,
+        buffer_offset: GPUSize64,
+        data: CustomAutoRooterGuard<ArrayBuffer>,
+        data_offset: GPUSize64,
+        size: Option<GPUSize64>,
+    ) -> Fallible<()> {
+        let bytes = data.to_vec();
+        let content_size = if let Some(s) = size {
+            s
+        } else {
+            bytes.len() as GPUSize64 - data_offset
```

Comment:
this should be taking the buffer size, not the bytes length

Reply:
The spec mentions it to be `data.byteLength` (https://gpuweb.github.io/gpuweb/#dom-gpuqueue-writebuffer). Should that be corrected?

Reply:
Oh wow, that's a surprise to me :) Thank you for correcting me!

Reply:
Filed https://github.com/gpuweb/gpuweb/pull/903 to clarify

Reply:
Should I change it to buffer size then?